### PR TITLE
Add static headers from annotation

### PR DESCRIPTION
**Pull request by WonderCsabo**
Refactor of #967. This is a WIP because it relies on #1448, so it cannot be merged yet. Otherwise it is done. The first interesting commit is 67249e9.

### CODE REVIEW

**Comment by WonderCsabo** [comment]
@yDelouis ready to be reviewed.

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/HeaderHandler.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		restValidatorHelper = new RestSpringValidatorHelper(environment, getTarget());
+	}
+
+	@Override
+	public void process(Element element, RestHolder holder) throws Exception {
+		// empty
+	}
+
+	@Override
+	protected void validate(Element element, ElementValidation validation) {
```
Comment (yDelouis) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/HeaderHandler.java]: You put `validate` after `process` again...

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestAnnotationHelper.java [hunk 2]
```diff
@@ -163,17 +165,40 @@ public boolean multipartHeaderRequired(ExecutableElement executableElement) {
...
+
+		Header header = executableElement.getAnnotation(Header.class);
+		if (header != null) {
+			processHeader(header, headerMap);
+		}
+
+		return headerMap;
+	}
+
+	private void processHeader(Header singleHeader, Map<String, String> headerMap) {
```
Comment (yDelouis) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestAnnotationHelper.java]: Are you sure you need a method for that ?

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/HeaderHandler.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		restValidatorHelper.elementHasOneOfRestMethodAnnotations(element, validation);
+
+		validatorHelper.doesNotHaveAnnotation(element, Headers.class, validation);
```
Comment (yDelouis) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/HeaderHandler.java]: Is this necessary ?

**Comment by yDelouis** [comment]
Seems good. I only have some small comments.

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/HeaderHandler.java (hunk 1 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/HeaderHandler.java]: Sorry, it was a messy rebase.

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/HeaderHandler.java (hunk 3 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/HeaderHandler.java]: We must define a precedence of Header and Headers and also write it in the doc if we remove this. Do we want to do that?

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestAnnotationHelper.java (hunk 2 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestAnnotationHelper.java]: Not really.

**Comment by WonderCsabo** [comment]
WIki updated.

**Pull request merged**